### PR TITLE
Repair http(s) includes after formatting

### LINKED ISSUE #2928: [BUG] Format adds spaces in urls with //
**Describe the bug**

I think format thinks urls have comments in it. So if you click format it adds a space before the //
**To Reproduce**

Right click and click format.

**Expected behavior**

Spaces aren't added to urls.
**Link**

before format ::
https://godbolt.org/z/x13vxoTnj
after format:
https://godbolt.org/z/rj7czceb9

**Screenshots**

**Desktop**

 - OS: 
 windows 10
 - Browser 
 chrome
 - Version 
93.0.4577.63
**Additional context**

### PR DESCRIPTION
**Pull request by falbrechtskirchinger**
clang-format inserts spaces after the URI scheme and before the hier-part. Remove the spaces to reform a valid URL.

For example, formatting the following code
```cpp
#include <https://localhost:8443/develop/json.hpp>
```
results in a broken https include:
```cpp
#include <https:  //localhost:8443/develop/json.hpp>
```

This could of course be done on the client side but might be a little messy. Let me know if that's preferable.

Fixes #2928.

### CODE REVIEW

**Comment by partouf** [comment]
Fixes https://github.com/compiler-explorer/compiler-explorer/issues/2928

I think this is an Ok solution to the problem

File: lib/formatters/clang-format.ts [hunk 1]
```diff
@@ -36,6 +36,16 @@ export class ClangFormatFormatter extends BaseFormatter {
...
+        return exec.execute(this.formatterInfo.exe, [`--style=${arg}`], {input: source}).then(result => {
+            if (result.code === 0) {
+                // Repair http(s) includes, i.e., remove the spaces inserted before the hier-part of the URI
+                const includeFind = /^(\s*#\s*include\s*["<]https?:)\s+(\/\/[^">]+[">].*)/;
+                const includeReplacement = '$1$2';
+                const lines = result.stdout.split('\n');
+                for (const idx in lines) lines[idx] = lines[idx].replace(includeFind, includeReplacement);
+                result.stdout = lines.join('\n');
+            }
+            return new Promise((resolve, _reject) => resolve(result));
```
Comment (matsjla) [inline lib/formatters/clang-format.ts:48]:
```suggestion
        const result = await exec.execute(this.formatterInfo.exe, [`--style=${arg}`], {input: source});
        if (result.code === 0) {
            // Repair http(s) includes, i.e., remove the spaces inserted before the hier-part of the URI
            const includeFind = /^(\s*#\s*include\s*["<]https?:)\s+(\/\/[^">]+[">].*)/;
            const includeReplacement = '$1$2';
            const lines = result.stdout.split('\n');
            for (const idx in lines) lines[idx] = lines[idx].replace(includeFind, includeReplacement);
            result.stdout = lines.join('\n');
        }
        return result;
```

**Review by matsjla — CHANGES_REQUESTED** [review]
I'm also fine with this fix.

I would however, want to replace the .then() into new Promise() chain with await. I've added a suggestion below. Otherwise LGTM! Thanks

**Review by matsjla — APPROVED** [review]
LGTM Thanks!

**Comment by mattgodbolt** [comment]
Thanks: this is now live!

**Pull request merged**